### ISSUE TITLE
Feature Request: Ability to Ignore Android Parceled Backstack

### ISSUE THREAD

**Issue by ScottPierce** [badoo/Decompose#84]
Instead of trying to restore the app state from `Parcelable` state, I'd like the ability to just ignore all parceled state and have my app always restart at my first screen no matter what. Is that possible?

**Comment by arkivanov** [comment]
Thanks, I will add it.

**Comment by arkivanov** [comment]
@ScottPierce After some deep investigation and attempts it appears to be very tricky, there are many corner cases and I find it very error prone.

However it is possible to disable back stack preservation using the following approach:
```kotlin
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val componentContext =
            DefaultComponentContext(
                lifecycle = lifecycle.asDecomposeLifecycle(),
                backPressedDispatcher = onBackPressedDispatcher.toBackPressedDispatcher()
            )

        val root = RootComponent(componentContext)

        setContent {
            // Render the root component
        }
    }
}
```

**Note 1:** this will disable state preservation as well. This should be fine because it does not make sense to preserve any data if the back stack is always reset.

**Note 2:** the back stack will reset even on configuration changes, and so `InstanceKeeper` will not retain anything

**Note 3:** I discovered a bug that the `InstanceKeeper` is not destroyed in case of Note 2. I will fix it in the next release.

**Comment by arkivanov** [comment]
Filed #86 for the Note 3

**Comment by arkivanov** [comment]
Let's keep this issue open for now, I will try to add something for the Note 2.

**Comment by arkivanov** [comment]
I addressed the Note 2 in #88. With the next release you will be able to do the following:

```kotlin
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val root =
            RootComponent(
                DefaultComponentContext(
                    lifecycle = lifecycle.asDecomposeLifecycle(),
                    stateKeeper = StateKeeper(savedStateRegistry) { isChangingConfigurations },
                    instanceKeeper = InstanceKeeper(viewModelStore),
                    backPressedDispatcher = BackPressedDispatcher(onBackPressedDispatcher)
                )
            )

        // Render the root
    }
}
```
